### PR TITLE
Replace deprecate With[Traces|Metrics|Logs][Receiver|Processor|Exporter] with the new definition

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

File: exporter/loggingexporter/factory.go [hunk 1]
```diff
@@ -37,8 +37,8 @@ func NewFactory() component.ExporterFactory {
 	return component.NewExporterFactory(
 		typeStr,
 		createDefaultConfig,
-		component.WithTracesExporterAndStabilityLevel(createTracesExporter, component.StabilityLevelInDevelopment),
-		component.WithMetricsExporterAndStabilityLevel(createMetricsExporter, component.StabilityLevelInDevelopment),
+		component.WithTracesExporter(createTracesExporter, component.StabilityLevelInDevelopment),
+		component.WithMetricsExporter(createMetricsExporter, component.StabilityLevelInDevelopment),
 		component.WithLogsExporterAndStabilityLevel(createLogsExporter, component.StabilityLevelInDevelopment),
```
Comment (codeboten) [inline exporter/loggingexporter/factory.go:42]: missed the logs

File: exporter/otlpexporter/factory.go [hunk 2]
```diff
@@ -35,8 +35,8 @@ func NewFactory() component.ExporterFactory {
 	return component.NewExporterFactory(
 		typeStr,
 		createDefaultConfig,
-		component.WithTracesExporterAndStabilityLevel(createTracesExporter, component.StabilityLevelStable),
-		component.WithMetricsExporterAndStabilityLevel(createMetricsExporter, component.StabilityLevelStable),
+		component.WithTracesExporter(createTracesExporter, component.StabilityLevelStable),
+		component.WithMetricsExporter(createMetricsExporter, component.StabilityLevelStable),
 		component.WithLogsExporterAndStabilityLevel(createLogsExporter, component.StabilityLevelBeta),
```
Comment (codeboten) [inline exporter/otlpexporter/factory.go:40]: Missed logs

File: exporter/otlphttpexporter/factory.go [hunk 3]
```diff
@@ -38,8 +38,8 @@ func NewFactory() component.ExporterFactory {
 	return component.NewExporterFactory(
 		typeStr,
 		createDefaultConfig,
-		component.WithTracesExporterAndStabilityLevel(createTracesExporter, component.StabilityLevelStable),
-		component.WithMetricsExporterAndStabilityLevel(createMetricsExporter, component.StabilityLevelStable),
+		component.WithTracesExporter(createTracesExporter, component.StabilityLevelStable),
+		component.WithMetricsExporter(createMetricsExporter, component.StabilityLevelStable),
 		component.WithLogsExporterAndStabilityLevel(createLogsExporter, component.StabilityLevelBeta),
```
Comment (codeboten) [inline exporter/otlphttpexporter/factory.go:43]: Missed logs

File: internal/testcomponents/example_exporter.go [hunk 4]
```diff
@@ -39,8 +39,8 @@ type ExampleExporterConfig struct {
 var ExampleExporterFactory = component.NewExporterFactory(
 	typeStr,
 	createExporterDefaultConfig,
-	component.WithTracesExporterAndStabilityLevel(createTracesExporter, stability),
-	component.WithMetricsExporterAndStabilityLevel(createMetricsExporter, stability),
+	component.WithTracesExporter(createTracesExporter, stability),
+	component.WithMetricsExporter(createMetricsExporter, stability),
 	component.WithLogsExporterAndStabilityLevel(createLogsExporter, stability),
```
Comment (codeboten) [inline internal/testcomponents/example_exporter.go:44]: Missed logs

**Review by codeboten — COMMENTED** [review]
Thanks for taking this on, just one comment (in multiple places 😅)

**Review by TylerHelmuth — COMMENTED** [review]
LGTM except for what @codeboten's pointed out already.

File: exporter/loggingexporter/factory.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/loggingexporter/factory.go]: Done.

File: exporter/otlpexporter/factory.go (hunk 2 above)
Comment (bogdandrutu) [reply exporter/otlpexporter/factory.go]: Done.

File: exporter/otlphttpexporter/factory.go (hunk 3 above)
Comment (bogdandrutu) [reply exporter/otlphttpexporter/factory.go]: Done.

File: internal/testcomponents/example_exporter.go (hunk 4 above)
Comment (bogdandrutu) [reply internal/testcomponents/example_exporter.go]: Done.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,9 +7,30 @@
...
+  - `component.WithTracesProcessor`
+  - `component.WithMetricsProcessor`
+  - `component.WithLogsProcessor`
 
 ### 🚩 Deprecations 🚩
 
+- Deprecate the following component functions added to ensure a stability level is set:
+  - `component.WithTracesExporterAndStabilityLevel` -> `component.WithTracesExporter`
+  - `component.WithMetricsExporterAndStabilityLevel` -> `component.WithMetricsExporter`
+  - `component.WithLogsExporterAndStabilityLevel` -> `component.WithLogsExporterl`
```
Comment (codeboten) [inline CHANGELOG.md:26]:
```suggestion
  - `component.WithLogsExporterAndStabilityLevel` -> `component.WithLogsExporter`
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,9 +7,30 @@
 - Remove deprecated funcs/types from service related to `Config` (#5755)
 - Change`confighttp.ToClient` to accept a `component.Host` (#5737)
 - Remove deprecated funcs from pdata related to mutable slices (#5754)
+- Change the following deprecated component functions to ensure a stability level is set:
+  - `component.WithTracesExporter`
+  - `component.WithMetricsExporter`
+  - `component.WithLogsExporterl`
```
Comment (codeboten) [inline CHANGELOG.md:13]:
```suggestion
  - `component.WithLogsExporter`
```

File: CHANGELOG.md (hunk 5 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 6 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: Done.

**Comment by bogdandrutu** [comment]
@codeboten my suggestion, when are minor comments like this, we should also approve the PR, or at least that is what I do to not wait for another round :)

**Review by codeboten — APPROVED** [review]

**Pull request merged**